compiler/src/iree/compiler/Dialect/Flow/Transforms/DumpDispatchGraph.cpp

```diff
@@ -0,0 +1,422 @@
+  /// cluster. Return the anchor node of the cluster.
+  Node emitClusterStmt(function_ref<void()> builder, std::string label = "") {
+    int clusterId = ++counter;
+    os << "subgraph cluster_" << clusterId << " {\n";
+    os.indent();
+    // Emit invisible anchor node from/to which arrows can be drawn.
+    Node anchorNode = emitNodeStmt(" ", kShapeNone);
+    os << attrStmt("label", quoteString(escapeString(std::move(label))))
+       << ";\n";
+    builder();
+    os.unindent();
+    os << "}\n";
+    return Node(anchorNode.id, clusterId);
+  }
+
+  /// Generate an attribute statement.
+  std::string attrStmt(const Twine &key, const Twine &value) {
+    return (key + " = " + value).str();
+  }
+
+  /// Emit an attribute list.
+  void emitAttrList(raw_ostream &os, const AttributeMap &map) {
+    os << "[";
+    interleaveComma(map, os, [&](const auto &it) {
+      os << this->attrStmt(it.getKey(), it.getValue());
+    });
+    os << "]";
+  }
+
+  // Print an MLIR attribute to `os`. Large attributes are truncated.
+  void emitMlirAttr(raw_ostream &os, Attribute attr) {
+    // A value used to elide large container attribute.
+    int64_t largeAttrLimit = getLargeAttributeSizeLimit();
+
+    // Always emit splat attributes.
+    if (attr.isa<SplatElementsAttr>()) {
+      attr.print(os);
+      return;
+    }
+
+    // Elide "big" elements attributes.
+    auto elements = attr.dyn_cast<ElementsAttr>();
+    if (elements && elements.getNumElements() > largeAttrLimit) {
+      os << std::string(elements.getType().getRank(), '[') << "..."
+         << std::string(elements.getType().getRank(), ']') << " : "
+         << elements.getType();
+      return;
+    }
+
+    auto array = attr.dyn_cast<ArrayAttr>();
+    if (array && static_cast<int64_t>(array.size()) > largeAttrLimit) {
+      os << "[...]";
+      return;
+    }
+
+    // Print all other attributes.
+    std::string buf;
+    llvm::raw_string_ostream ss(buf);
+    attr.print(ss);
+    os << truncateString(ss.str());
+  }
+
+  /// Append an edge to the list of edges.
+  /// Note: Edges are written to the output stream via `emitAllEdgeStmts`.
+  void emitEdgeStmt(Node n1, Node n2, std::string label, StringRef style) {
+    AttributeMap attrs;
+    attrs["style"] = style.str();
+    // Do not label edges that start/end at a cluster boundary. Such edges are
+    // clipped at the boundary, but labels are not. This can lead to labels
+    // floating around without any edge next to them.
+    if (!n1.clusterId && !n2.clusterId)
+      attrs["label"] = quoteString(escapeString(std::move(label)));
+    // Use `ltail` and `lhead` to draw edges between clusters.
+    if (n1.clusterId)
+      attrs["ltail"] = "cluster_" + std::to_string(*n1.clusterId);
+    if (n2.clusterId)
+      attrs["lhead"] = "cluster_" + std::to_string(*n2.clusterId);
+
+    edges.push_back(strFromOs([&](raw_ostream &os) {
+      os << llvm::format("v%i -> v%i ", n1.id, n2.id);
+      emitAttrList(os, attrs);
```

Comment:
I see from the example that there is some number printed on the edges. Cant figure out from code what those numbers are

Reply:
It is the operand index number. For dispatches, it always starts from 3 though.